velodyne_gazebo_plugins/src/GazeboRosVelodyneLaser.cpp

```diff
@@ -365,22 +349,83 @@ void GazeboRosVelodyneLaser::OnScan(ConstLaserScanStampedPtr& _msg)
   msg.fields[4].datatype = sensor_msgs::msg::PointField::UINT16;
   msg.fields[4].count = 1;
   msg.data.resize(scan_pattern_.horizontal_samples.size() *
-                  scan_pattern_.horizontal_samples.size() *
+                  scan_pattern_.vertical_samples.size() *
                   POINT_STEP);
 
+  using ScanPattern = GazeboRosVelodyneLaser::ScanPattern;
   uint8_t *ptr = msg.data.data();
-  for (size_t i = 0; i < scan_pattern_.horizontal_samples.size(); i++) {
-    for (size_t j = 0; j < scan_pattern_.vertical_samples.size(); j++) {
-      int ray_index = scan_pattern_.horizontal_samples[i].index +
-        scan_pattern_.vertical_samples[j].index * scan_pattern_.horizontal_samples.size();
-
-      // Range
-      double r = _msg->scan().ranges(ray_index);
-      // Intensity
-      double intensity = _msg->scan().intensities(ray_index);
+  for (size_t i = 0; i < scan_pattern_.horizontal_samples.size(); ++i) {
+    const ScanPattern::Sample & hsample = scan_pattern_.horizontal_samples[i];
+
+    for (size_t j = 0; j < scan_pattern_.vertical_samples.size(); ++j) {
+      const ScanPattern::Sample & vsample = scan_pattern_.vertical_samples[j];
+
+      double r = INFINITY;
```

Comment:
I think this stands for `range`; can we use that slightly more descriptive name?

Reply:
Sounds good, I was keeping preexisting naming but I agree it can be made better. See c73e6e2ac4b017822ea824ff94e082533aeed578.